src/main/resources/avro/genotypephenotype.avdl

```diff
@@ -0,0 +1,138 @@
+@namespace("org.ga4gh")
+/**
+
+This protocol defines the associations between genotype
+and phenotype (G2P).  Associations can be made as a
+result of literature curation, computational modeling,
+inference, etc., and modeled and shared using this schema.
+
+Here, we follow the dogma of:
+
+      Genotype + Environment = Phenotype
+
+where a G2P association is between the G(enotype) in the context of
+some E(environment), which gives rise to a P(henotype). These
+associations have further evidence, provenance, and attribution.
+
+We leverage the GenomicFeature in the sequenceAnnotation schema here
+as it can accomodate any genomic feature from a single nucleotide variation
+(SNV), up through a gene, and/or complex rearrangements.  Each can
+be modeled as genomic features, and generally linked to a phenotype.
+Collections of these features can represent a genotype at different levels
+of completeness.  Therefore, we can represent single allelic variation,
+allelic complement, and multiple variants in a genotype that can each or
+collectively be associated with a phenotype.
+
+To enable standardized integration, this schema relies heavily on
+OntologyTerms, both for typing phenotype (for example, using the HPO or DO)
+but also genomic features (such as using SO) and levels of evidence
+(ECO).
+
+*/
+protocol GenotypePhenotype {
+
+import idl "common.avdl";
+import idl "metadata.avdl";
+import idl "sequenceAnnotations.avdl";
+
+
+/**
+The context in which a genotype gives rise to a phenotype.
+This is fairly open-ended; as a stub we have a simple ontology term.
+For example, a controlled term for a drug
+*/
+record EnvironmentalContext {
+
+  /** TODO: define potential ontologies */
+  org.ga4gh.models.OntologyTerm environmentType;
+
+  /** A textual description of the environment. This is used to complement the structured description in the environmentType field */
```

Comment:
This line is too long. In the Github diff view it is cut off at the right around "in the enviro", and the horizontal scroll bar is all the way at the bottom of the view for the file.